src/ServarrAPI/Controllers/UpdateController.cs

```diff
@@ -13,23 +14,28 @@ namespace ServarrAPI.Controllers.Update
     [Route("[controller]")]
     public class UpdateController : Controller
     {
+        private readonly Config _config;
         private readonly IUpdateFileService _updateFileService;
 
-        public UpdateController(IUpdateFileService updateFileService)
+        public UpdateController(IUpdateFileService updateFileService, IOptions<Config> optionsConfig)
         {
             _updateFileService = updateFileService;
+            _config = optionsConfig.Value;
         }
 
         [Route("{branch}/changes")]
         [HttpGet]
         public async Task<object> GetChanges([FromRoute(Name = "branch")] string updateBranch,
+                                             [FromQuery(Name = "version")] string urlVersion,
                                              [FromQuery(Name = "os")] OperatingSystem operatingSystem,
                                              [FromQuery(Name = "runtime")] Runtime runtime = Runtime.DotNet,
                                              [FromQuery(Name = "arch")] Architecture arch = Architecture.X64)
         {
             Response.Headers[HeaderNames.CacheControl] = GetCacheControlHeader(DateTime.UtcNow.AddDays(30));
 
-            var updateFiles = await _updateFileService.Find(updateBranch, operatingSystem, runtime, arch, 5);
+            var maxVersion = GetMaxVersion(urlVersion);
```

Comment:
I wonder if `UpdateFileService` should handle the calculation of `maxVersion` to keep the logic out of the controllers